### PR TITLE
Reduce dependencies for chatops controller

### PR DESCRIPTION
**Pull request by dbussink**
When looking at options to reduce dependencies for a Rails app, my approach failed because we can't drop some Rails components because we use gems that express all of Rails as a dependency instead of the parts they need.

The chatops controller is one of those. It looks like though it only needs ActiveSupport & ActionPack so I reduced the dependencies here to just those parts.

### CODE REVIEW

**Comment by dbussink** [comment]
@leequarella Does it make sense to pull this into #54 instead of this PR? The dependency change I mean, since you already did all the other bits in a better way?

File: chatops-controller.gemspec [hunk 1]
```diff
@@ -17,7 +17,8 @@ Gem::Specification.new do |s|
   s.files = Dir["{app,config,db,lib}/**/*", "README.md"]
   s.test_files = Dir["spec/**/*"]
 
-  s.add_dependency "rails", ">= 4.0"
+  s.add_dependency "actionpack", ">= 4.0"
+  s.add_dependency "activesupport", ">= 4.0"
```
Comment (parkr) [inline chatops-controller.gemspec:21]: ❤️ We're also looking for this over in github/cat!

File: spec/lib/chatops/controller_spec.rb [hunk 2]
```diff
@@ -60,7 +60,7 @@ def ensure_app_given
   end
 
   def rails_flexible_post(path, outer_params, jsonrpc_params = nil)
-    if Rails.version.starts_with?("4")
+    if Rails.version.start_with?("4")
```
Comment (parkr) [inline spec/lib/chatops/controller_spec.rb:63]: `if defined?(Rails)` necessary here?

**Comment by mistydemeo** [comment]
@dbussink Gentle reminder this is a public repo. 😄

File: spec/lib/chatops/controller_spec.rb (hunk 2 above)
Comment (mistydemeo) [reply spec/lib/chatops/controller_spec.rb:63]: From checking the ActionPack source, it appears `Rails` will be defined even if all of Rails isn't loaded. https://github.com/rails/rails/blob/main/actionpack/lib/action_dispatch/railtie.rb#L7

**Comment by dbussink** [comment]
Was included in #54

**Pull request closed without merging**